DuckDuckGo/Tab/View/BrowserTabViewController.swift

```diff
@@ -334,8 +349,59 @@ final class BrowserTabViewController: NSViewController {
     private func addWebViewToViewHierarchy(_ webView: WebView, tab: Tab) {
         let container = WebViewContainerView(tab: tab, webView: webView, frame: view.bounds)
         self.webViewContainer = container
+        containerStackView.orientation = .vertical
+        containerStackView.distribution = .fill
+        containerStackView.alignment = .leading
+        containerStackView.distribution = .fillProportionally
+
         // Make sure link preview (tooltip shown in the bottom-left) is on top
-        view.addSubview(container, positioned: .below, relativeTo: hoverLabelContainer)
+        view.addSubview(containerStackView, positioned: .below, relativeTo: hoverLabelContainer)
+
+        containerStackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            containerStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            containerStackView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            containerStackView.topAnchor.constraint(equalTo: view.topAnchor),
+            containerStackView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+        containerStackView.addArrangedSubview(container)
+    }
+    var daxContextualOnboardingController: NSViewController?
+
+    private func presentContextualOnboarding() {
+        // Before presenting a new dialog, remove any existing ones.
+        containerStackView.arrangedSubviews.filter({ $0 != webViewContainer }).forEach {
+            containerStackView.removeArrangedSubview($0)
+            $0.removeFromSuperview()
+        }
+
+        guard featureFlagger.isFeatureOn(.highlightsOnboarding) else { return }
+
+        guard let tab = tabViewModel?.tab,
+              let dialogType = onboardingDialogTypeProvider.dialogTypeForTab(tab) else {
+            return
+        }
+
+        var onDismissAction: () -> Void = {}
+        if let webViewContainer {
+            onDismissAction = {
```

Comment:
This captures BrowserTabViewController. when you clear data with the fire button deinit does not get called so the DeallocationCheck throws an assertion because the WebView leaked. Adding [weak self] will break the retain cycle.

<img width="1203" alt="Screenshot 2024-09-06 at 4 59 21 PM" src="https://github.com/user-attachments/assets/ec94783a-5dd9-4112-b89e-48cd4d56d11f">